pkg/types/chain_writer.go

```diff
@@ -0,0 +1,55 @@
+package types
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/google/uuid"
+)
+
+type ChainWriter interface {
+	// SubmitTransaction packs and broadcasts a transaction to the underlying chain.
+	//
+	// The `payload` array must contain the encoded transaction payload, and the related signatures.
+	// The `transactionID` will be used by the underlying TXM as an idempotency key, and unique reference to track transaction attempts.
+	SubmitTransaction(ctx context.Context, payload []any, transactionID uuid.UUID, toAddress string, meta *TxMeta, value big.Int) (int64, error)
```

Comment:
Why is the payload a slice of `any` instead of just `any`?

Reply:
This has been updated after some discussions with @patrick-dowell.